src/bandersnatch/tests/test_utils.py

```diff
@@ -60,16 +62,16 @@ def test_find_files():
             with afile.open("w") as afp:
                 afp.write("PyPA ftw!")
 
-        found_files = set()
+        found_files: Set[Path] = set()
         recursive_find_files(found_files, td_path)
         assert found_files == expected_found_files
 
 
-def test_rewrite(tmpdir, monkeypatch):
+def test_rewrite(tmpdir: Path, monkeypatch: MonkeyPatch) -> None:
     monkeypatch.chdir(tmpdir)
     with open("sample", "w") as f:
         f.write("bsdf")
-    with rewrite("sample") as f:
+    with rewrite("sample") as f:  # type: ignore
```

Comment:
What is the type error here ... weird. I’ll look after merge.

Reply:
The error is:
```py
src/bandersnatch/tests/test_utils.py: note: In function "test_rewrite":
src/bandersnatch/tests/test_utils.py:74: error: Incompatible types in
assignment (expression has type "IO[Any]", variable has type "TextIO")
        with rewrite("sample") as f:
```

Reply:
The problem is that in the open() call above has the return type `typing.TextIO` because mypy sees the `"w"` and returns a more specific type. The rewrite function can take any mode and then pass it along to `tempfile.NamedTemporaryFile`. Since the mode is defined by a variable, mypy just sets the return type of `tempfile.NamedTemporaryFile` and subsequently `rewrite` to `typing.IO[Any]`.¹

A solution that comes to mind is just adding a `f: IO[Any]` type annotation before the first with statement so `f` isn't inferred as a subclass (`typing.TextIO`) of `typing.IO`.

---
¹. Quoting from Guido van Rossum https://github.com/python/typeshed/issues/1229#issuecomment-304932714

Reply:
Can't trust that guy, but that's what I would have tried too. Thanks @ichard26